src/simplex.pybind.h

```diff
@@ -70,27 +71,145 @@ void add_simplex_module(py::module& root) {
           "get_observables_from_errors",
           [](SimplexDecoder& self, const std::vector<size_t>& predicted_errors) {
             std::vector<bool> result(self.num_observables, false);
-            const auto& indices = self.get_flipped_observables(predicted_errors);
-            for (int index : indices) {
-              result[index] = true;
+            for (size_t ei : predicted_errors) {
+              for (int obs_index : self.errors[ei].symptom.observables) {
+                result[obs_index] = result[obs_index] ^ true;
+              }
             }
             return result;
           },
           py::arg("predicted_errors"))
       .def("cost_from_errors", &SimplexDecoder::cost_from_errors, py::arg("predicted_errors"))
       .def(
-          "decode",
+          "decode_from_detection_events",
           [](SimplexDecoder& self, const std::vector<uint64_t>& detections) {
-            std::vector<bool> result(self.num_observables, false);
+            std::vector<char> result(self.num_observables, false);
             self.decode(detections);
             for (size_t ei : self.predicted_errors_buffer) {
               for (int obs_index : self.errors[ei].symptom.observables) {
                 result[obs_index] = result[obs_index] ^ true;
               }
             }
-            return result;
+            return py::array(py::dtype::of<bool>(), result.size(), result.data());
           },
           py::arg("detections"),
-          py::call_guard<py::scoped_ostream_redirect, py::scoped_estream_redirect>());
+          py::call_guard<py::scoped_ostream_redirect, py::scoped_estream_redirect>(),
+          R"pbdoc(
+          Decodes a single shot from a list of detection events.
+
+          Parameters
+          ----------
+          detections : list[int]
+              A list of `uint64_t` indices corresponding to the detectors that were
```

Comment:
Maybe remove this uint64_t type since the docstring is viewed in python
```suggestion
              A list of indices corresponding to the detectors that were
```